backend/api/Utilities/TagPositioner.cs

```diff
@@ -13,10 +14,24 @@ public class TagPositioner
             Justification = "Will be implemented, is here for documentation of intended behavior")]
         public static IsarPose GetPoseFromTag(EchoTag tag)
         {
-            const string Frame = "asset";
-            IsarPosition position = new(0, 0, 0, Frame);
-            IsarOrientation orientation = new(0, 0, 0, 1, Frame);
-            return new IsarPose(position, orientation, Frame);
+            using var r = new StreamReader("./Utilities/PredefinedPoses.json");
+            string json = r.ReadToEnd();
+            var predefinedPoses = JsonSerializer.Deserialize<List<PredefinedPose>>(json);
+
+            if (predefinedPoses == null)
+                throw new RobotPositionNotFoundException("Could not find any predefined poses in the workaround file");
+
+
+            var predefinedPose = predefinedPoses.Find(x => x.Tag == tag.TagId);
+
+            if (predefinedPose == null)
+                throw new RobotPositionNotFoundException($"Could not find tag {tag.TagId} in the workaround file");
+
+            IsarPosition position = new(predefinedPose.Pose.Position.X, predefinedPose.Pose.Position.Y, predefinedPose.Pose.Position.Z, predefinedPose.Pose.Frame);
+            IsarOrientation orientation = new(predefinedPose.Pose.Orientation.X, predefinedPose.Pose.Orientation.Y, predefinedPose.Pose.Orientation.Z, predefinedPose.Pose.Orientation.W, predefinedPose.Pose.Frame);
+
+            return new IsarPose(position, orientation, predefinedPose.Pose.Frame);
```

Comment:
Suggesting implementing a constructor for `IsarPose` taking a `PredefinedPose` as argument instead of this.